packages/components/src/DataTable/Item/DataTableRow.tsx

```diff
@@ -119,6 +121,18 @@ const DataTableRowLayout = forwardRef(
 
 DataTableRowLayout.displayName = 'DataTableRowLayout'
 
+const getRowHoverColor = (
+  theme: DefaultTheme,
+  hasOnClick: boolean,
+  isHeader: boolean | undefined,
+  isSelected: boolean | 'mixed' | undefined
```

Comment:
See note below about tightening up inputs, it's a lot easier to test when a functions parameters are all required:

```suggestion
  isHeader: boolean,
  isSelected: boolean
```